src/maxreg.rs

```diff
@@ -0,0 +1,135 @@
+use crate::traits::{CmRDT, CvRDT};
+use serde::{Deserialize, Serialize};
+use std::{error, fmt};
+
+/// `MaxReg` Holds a monotonically increasing value that implements the Ord trait. For use of floating-point values,
+/// you must create a wrapper (or use a crate like `float-ord`)
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct MaxReg<V: Ord + Copy> {
```

Comment:
This constraint of `Copy` values seems a bit too restrictive.

Also, we should avoid putting trait constraints on the struct definition, only put the constraints on the `impl` blocks where they are needed.

e.g. for the `Default` trait implementation we don't need the Ord/Copy constraint on `V`, we only need `V: Default`.

Reply:
Hi @davidrusu , sorry for the delay; was at a workshop and now I am back. I will make this change. I only did it because rust was throwing an error in the `CmRDT` implementation if I did not include it:
```rust
the trait bound `V: std::marker::Copy` is not satisfied
the trait `std::marker::Copy` is not implemented for `V`
```